applications/CompressiblePotentialFlowApplication/custom_processes/define_2d_wake_process.cpp

```diff
@@ -0,0 +1,326 @@
+{
+    ModelPart& root_model_part = mrBodyModelPart.GetRootModelPart();
+
+    // Initialize element variables
+    VariableUtils().SetNonHistoricalVariable(WAKE, false, root_model_part.Elements());
+    VariableUtils().SetNonHistoricalVariable(KUTTA, false, root_model_part.Elements());
+    VariableUtils().SetNonHistoricalVariable(TRAILING_EDGE, false, root_model_part.Elements());
+    VariableUtils().SetFlag(STRUCTURE, false, root_model_part.Elements());
+    VariableUtils().SetNonHistoricalVariable(ELEMENTAL_DISTANCES, ZeroVector(3), root_model_part.Elements());
+
+    // Initialize node variables
+    VariableUtils().SetNonHistoricalVariable(TRAILING_EDGE, false, root_model_part.Nodes());
+    VariableUtils().SetVariable(DISTANCE, 1.0, root_model_part.Nodes());
+}
+
+// This function sets the wake direction and normal for further computations
+const void Define2DWakeProcess::SetWakeDirectionAndNormal()
+{
+    // Reading the free_stream_velocity from the properties
+    const auto free_stream_velocity = mrBodyModelPart.GetProcessInfo().GetValue(FREE_STREAM_VELOCITY);
+    KRATOS_ERROR_IF(free_stream_velocity.size() != 3)
+        << "The free stream velocity should be a vector with 3 components!"
+        << std::endl;
+
+    // Computing the norm of the free_stream_velocity vector
+    const double norm = std::sqrt(inner_prod(free_stream_velocity, free_stream_velocity));
+
+    KRATOS_ERROR_IF(norm == 0.0)
+        << "The norm of the free stream velocity should be different than 0."
+        << std::endl;
+
+    // The wake direction is the free stream direction
+    mWakeDirection = free_stream_velocity / norm;
+
+    // Rotating 90° to obtain the wake normal
+    mWakeNormal(0) = -mWakeDirection(1);
+    mWakeNormal(1) = mWakeDirection(0);
+    mWakeNormal(2) = 0.0;
+}
+
+// This function finds and saves the trailing edge for further computations
+const void Define2DWakeProcess::SaveTrailingEdgeNode()
+{
+    double max_x_coordinate = -std::numeric_limits<double>::max();;
+    NodeIteratorType trailing_edge_node;
+    for (auto it_node = mrBodyModelPart.NodesBegin();
+         it_node != mrBodyModelPart.NodesEnd(); ++it_node) {
+        if (it_node->X() > max_x_coordinate) {
+            max_x_coordinate = it_node->X();
+            trailing_edge_node = it_node;
+        }
+    }
+    trailing_edge_node->SetValue(TRAILING_EDGE, true);
+    mTrailingEdgeNode = trailing_edge_node;
+}
+
+// This function checks which elements are cut by the wake and marks them as
+// wake elements
+const void Define2DWakeProcess::MarkWakeElements()
+{
+    ModelPart& root_model_part = mrBodyModelPart.GetRootModelPart();
+    bool potentially_wake = false;
+    bool is_wake_element = false;
+    BoundedVector<double, 3> nodal_distances_to_wake = ZeroVector(3);
+
+    const int number_of_threads = OpenMPUtils::GetNumThreads();
+    std::vector<std::unordered_set<std::size_t>> all_threads_trailing_edge_element_ids(number_of_threads);
```

Comment:
Maybe we can define a 

`typedef std::vector<std::unordered_set<std::size_t>> UnorderedSetType`

in the header, to avoid long names